include/copc-lib/io/writer.hpp

```diff
@@ -73,7 +73,9 @@ class Writer : public BaseIO
         uint64_t points_by_return_14[15]{};
     };
 
-    Writer(std::ostream &out_stream, LasConfig const &config, int span = 0, const std::string &wkt = "");
+    Writer(std::ostream *out_stream, LasConfig const &config, int span = 0, const std::string &wkt = "");
+
+    Writer(const std::string &file_path, LasConfig const &config, int span = 0, const std::string &wkt = "");
```

Comment:
I would make a subclass of Writer. Then, do the fstream creation and destruction there. WriterInternal should be agonistic to those details.